src/thinkspace/api/thinkspace-readiness-assurance/app/models/thinkspace/readiness_assurance/assessment.rb

```diff
@@ -29,14 +29,109 @@ def sync_assessments
         phase      = self.authable
         assignment = Thinkspace::Casespace::Assignment.find(phase.assignment_id)
 
-        if assignment.sync_rat
+        if assignment.sync_rat?
           assessments = Thinkspace::ReadinessAssurance::Assessment.where(authable: assignment.thinkspace_casespace_phases).without(self)
           assessments.each do |assessment|
             assessment.update_columns(questions: self.questions, answers: self.answers, settings: self.settings.deep_merge(assessment.settings))
           end
         end
       end
 
+      def sync(options)
+        phase      = self.authable
+        assignment = Thinkspace::Casespace::Assignment.find(phase.assignment_id)
+
+        scoring_keys  = ['correct', 'attempted', 'no_answer', 'incorrect_attempt']
+        question_keys = ['type', 'random', 'ifat', 'justification']
+
+        if assignment.sync_rat?
```

Comment:
This is some pretty complicated logic.  Couple of notes:
  * It'd be nice to pull this out into a class that can not pollute the model as much.  There's a lot of nested if/if statements that would be nice to have as separate method (which lends well towards the class abstraction).  For example, you could have a method called `sync_settings` which handles just the settings, etc.
  * I'd use `dig` pretty much everywhere.  It just feels a lot safer than running into `undefined [] for nil` type of error.  I know most of these are scoped out by the if wrappers.